pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -96,11 +101,19 @@ func NewCloudProvider(ctx context.Context, options cloudprovider.Options) *Cloud
 				getCABundle(ctx),
 			),
 		},
+		kubeClient: options.KubeClient,
 	}
 }
 
 // Create a node given the constraints.
 func (c *CloudProvider) Create(ctx context.Context, nodeRequest *cloudprovider.NodeRequest) (*v1.Node, error) {
+	if nodeRequest.Template.ProviderRef != nil {
+		aws, err := c.getProvider(ctx, nodeRequest.Template.ProviderRef.Name)
```

Comment:
Thoughts on encapsulating `if nodeRequest.Template.ProviderRef != nil {` inside c.getProvider?

You've duplicated some logic around 
`c.instanceProvider.Create(ctx, aws, nodeRequest)`

Reply:
Essentially, c.getProvider returns you an `*AWS` from the node template, and the caller figures out what to do with it.